notifier/main.ts

```diff
@@ -1,12 +1,67 @@
 import * as path from "std/path/mod.ts";
-import { Application } from "oak/mod.ts";
-import { Bot } from "grammy/mod.ts";
+import { Application, Router } from "oak/mod.ts";
+import { Bot, Context, webhookCallback } from "grammy/mod.ts";
+import { FileFlavor, hydrateFiles } from "grammy_files/mod.ts";
 import env from "./env.ts";
 import * as db from "./db.ts";
 import { verifyGitHubWebhook } from "./utils.ts";
 
 const app = new Application();
-const bot = new Bot(env.BOT_TOKEN);
+const router = new Router();
+const bot = new Bot<FileFlavor<Context>>(env.BOT_TOKEN);
+
+bot.api.config.use(hydrateFiles(bot.token));
+
+bot.chatType(["group", "supergroup"]).filter((ctx) =>
+  ctx.chat.id == env.CHAT_ID
+).on("message:document", async (ctx) => {
+  if (
+    !ctx.message.document.file_name?.endsWith(".patch") || !ctx.message.caption
+  ) {
+    return;
+  }
+  const repliedMessage = ctx.message.reply_to_message;
+  if (
+    !repliedMessage || !(repliedMessage.from?.id == ctx.me.id)
+  ) {
+    return;
+  }
+  const prNumber = Number(
+    (repliedMessage.text ?? "").match(/^#([1-9][0-9]+)\s/)?.[1],
+  );
```

Comment:
How about parsing this from the message entities in order to avoid the regex?

Reply:
Alright.